### ISSUE TITLE
Make Source owned

### ISSUE THREAD

**Issue by RazrFalcon** [jrmuizel/raqote#50]
Currently, it's pretty hard to use it. Let's say I want to create a pattern in a function:

```rust
fn create_pattern<'a>() -> Source<'a> {}
    let dt = DrawTarget::new(10, 10);

    let img = Image {
        width: dt.width(),
        height: dt.height(),
        data: dt.get_data(),
    };

    // I can't do this, because `dt` doesn't live long enough.
    Source::Image(
        img,
        ExtendMode::Repeat,
        Transform::default(),
    )
}
```

So I have to create `dt` outside this method, which is not user friendly.

Another example:

```rust
if use_pattern {
    let dt = DrawTarget::new(10, 10);
    ... // `dt` doesn't live long enough again
}
```

I understand that copying is expensive, but maybe we should use `Rc` for `Source::Image` or something like this.

**Comment by jrmuizel** [comment]
Yeah, I don't think what we currently have is great. What I'd recommend for now is only build up the Source object when you're about to do an operation and have the image data owned by some other object. I've added a DrawTarget::into_vec() method that gives you the underlying Vec which should help with this a little.

**Comment by RazrFalcon** [comment]
I think this can be fixed by extracting the `Image` part from `DrawTarget`. Which is basically #26.

>DrawTarget::into_vec()

I think `as_image` will be even better =)
